code/src/components/AddTodo.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react'; 
+import { useDispatch } from 'react-redux';
+
+import { todos } from '../reducers/todos';
+import { AddTodoForm, NewTodo, TodoText, AddButton } from './StyledComponents/AddTodoStyling';
+
+export const AddTodo = () => { 
+  const dispatch = useDispatch();
+  const [addTodo, setAddTodo] = useState('');
+
+  const handleSubmit = event => {
+    event.preventDefault()
+    dispatch(
+      todos.actions.addTodo(
+        {
+          id: Date.now(),
+          text: addTodo,
+          complete:false,
+        }
+      )
+    );
+    setAddTodo ('');
+  };
+
+  // adds users input to the to-do list by pushing (push()) data into the array, defined in the
+  //todos.js reducer
+  return (
+    <AddTodoForm onSubmit={handleSubmit}>
+      <NewTodo>
+      <TodoText 
+        type='text' required
+        value={addTodo}
+        placeholder='Add a to-do!'
+        onChange={(event)=> setAddTodo(event.target.value)}
+      />
+        <AddButton type='submit' disabled={ addTodo.length < 5 }> + </AddButton>
```

Comment:
The button disabled is a great touch! Food for thought - is 5 the best number? or could it maybe be a little bit lower?

Reply:
I can't really see that it's disabled though... Add some styling or some other visual indicator to the user that the todo cannot be submitted

Reply:
Thank you! I just fixed the colour on disabled, so now it's greyish until the user reaches the 5 chars and then turns pink = did this by adding props! learned a new thing today! Also changed the hover colour (which was a grey 😅) so it doesn't get mixed up with the disabled status!